libclamav_rust/src/logging.rs

```diff
@@ -81,3 +84,9 @@ mod tests {
         error!("friend.");
     }
 }
+
+#[no_mangle]
+pub extern "C" fn clrs_eprint(c_buf: *const c_char) -> () {
```

Comment:
This function needs some documentation, at least with respect to its intent.  It looks like you're trying to avail yourself of the Rust standard library's locking around writing to `stderr`, and pretty-printing any non-UTF8 characters as �. But, unless you're consistent throughout the code at never utilizing the C library's `fprintf(stderr, ...)`, you'll still end up intermingling output since the two implementations use separate buffers.

Also need (minimally) a NULL check on `c_buf`.

Reply:
I'll check if `fprintf(stderr` is used within scans outside of our logging functions.  If there is other use of `fprintf(stderr` and it's not a simple fix, I may just back-off the "[libclamav: fix utf8 logging issues on Windows](https://github.com/Cisco-Talos/clamav/pull/549/commits/cbf3f592b11e184bd8952489e69cd9b911c4724f)" commit.

Reply:
> Also need (minimally) a NULL check on c_buf.

👍 good point.